src/gis/components/MapLoader.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2023 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { CircularProgress, Paper, Stack } from '@mui/material';
+
+const MapLoader = props => {
```

Comment:
question: do we have other loaders? Can/should this be a general-purpose loader?

Reply:
@paulschreiber This one has the same strcuture of the map, so that the layout doesn't change while it is loading, I think it might be a good idea to have specific loaders for some components and some general ones for other uses, but I think that falls under UX decisions